File: backend/internal/domain/enrollment.go
package domain

import (
	"time"
)

// Status values: QUEUING, SUCCESS, FAILED, CANCELLED
type Enrollment struct {
	ID            uint64     `gorm:"primaryKey" json:"id"`
	UserID        uint64     `gorm:"not null;uniqueIndex:idx_user_activity" json:"user_id"`
	ActivityID    uint64     `gorm:"not null;uniqueIndex:idx_user_activity" json:"activity_id"`
	Status        string     `gorm:"type:varchar(20);not null;default:'QUEUING'" json:"status"`
	QueuePosition *int       `json:"queue_position"`
	EnrolledAt    time.Time  `gorm:"not null" json:"enrolled_at"`
	FinalizedAt   *time.Time `json:"finalized_at"`
}

File: frontend/src/types/enrollment.ts
export type EnrollmentStatus = 'QUEUING' | 'SUCCESS' | 'FAILED' | 'CANCELLED';

export interface CreateEnrollmentResult {
  code: number;
  message: string;
  status: EnrollmentStatus;
  enrollmentId?: number;
  activityId?: number;
  queuePosition?: number;
  estimatedWaitSeconds?: number;
  stockRemaining?: number;
  orderNo?: string;
}

export interface EnrollmentStatusDetail {
  enrollmentId: number;
  activityId: number;
  status: EnrollmentStatus;
  submittedAt: string;
  activityTitle?: string;
  orderNo?: string;
  finalizedAt?: string;
  queuePosition?: number;
  estimatedWaitSeconds?: number;
}

export interface EnrollmentListItem {
  id: number;
  userId: number;
  activityId: number;
  status: EnrollmentStatus;
  queuePosition?: number | null;
  enrolledAt: string;
  finalizedAt?: string | null;
}

export interface EnrollmentListResult {
  list: EnrollmentListItem[];
  total: number;
  page: number;
  pageSize: number;
}

export interface EnrollmentStatusItem {
  enrollmentId: number;
  activityId: number;
  activityTitle?: string;
  status: EnrollmentStatus;
  queuePosition?: number;
  estimatedWaitSeconds?: number;
  submittedAt?: string;
  finalizedAt?: string;
  orderNo?: string;
}

export interface CancelEnrollmentResult {
  enrollmentId: number;
  status: EnrollmentStatus;
}
